web_src/js/features/projects.js

```diff
@@ -1,56 +1,125 @@
 
-  for (const column of boardColumns) {
-    new Sortable(
-      column.getElementsByClassName('board')[0],
-      {
-        group: 'shared',
-        animation: 150,
-        onAdd: (e) => {
-          $.ajax(`${e.to.dataset.url}/${e.item.dataset.issue}`, {
-            headers: {
-              'X-Csrf-Token': csrf,
-              'X-Remote': true,
-            },
-            contentType: 'application/json',
-            type: 'POST',
-            error: () => {
-              e.from.insertBefore(e.item, e.from.children[e.oldIndex]);
-            },
-          });
-        },
+  const colContainer = document.getElementById('board-container');
+  let projectURL = '';
+  let projectID = 0;
+  if (colContainer && colContainer.dataset) {
+    projectURL = colContainer.dataset.url;
+    projectID = colContainer.dataset.projectid;
+  }
+
+  if (colContainer) {
+    $('body').keyup((e) => {
+      if (e.keyCode === 27) {
+        $('#current-card-details').hide();
+        $('#current-card-details').html('');
+        $('#current-card-details-input').html('');
       }
-    );
+    });
   }
 
-  $('.edit-project-board').each(function () {
-    const projectTitleLabel = $(this).closest('.board-column-header').find('.board-label');
+  // search as you type new issues
+  $('#current-card-details-input').on('keyup', '#issue-search', (e) => {
+    const q = e.currentTarget.value;
+    const repoURL = $('[data-repourl]').data('repourl');
+    fetch(`/api/v1/repos${repoURL}/issues?state=open&q=${q}&exclude_project_id=${projectID}`, {
+      method: 'GET',
+      headers: {'X-Csrf-Token': csrf, 'Content-Type': 'application/json'}
+    })
+      .then((res) => {
+        return res.json();
+      })
+      .then((data) => {
+        let cards = '';
+        data.map((issue) => {
+          const card = `
+            <div class='card draggable-card board-card' data-priority='0' data-id='0' data-issueid='${issue.id}'>
+              <div class='content'>
+                <div class='header'>
+                  <a href='${repoURL}/${issue.IsPull ? 'pulls' : 'issues'}/${issue.number}/sidebar/true' data-url='${repoURL}/${issue.IsPull ? 'pulls' : 'issues'}/${issue.number}/sidebar/true'>#${issue.number} ${issue.title}</a>
+                </div>
+                <div class='meta'></div>
+                <div class='description'></div>
+              </div>
+            </div>
+          `;
+          cards += card;
+        });
+        $('#issue-results').html(
+          `<div class='ui cards draggable-cards'>${cards}</div>`
+        );
+        sortCards('.draggable-cards');
+      });
+  });
+  sortCards('.draggable-cards');
+
+  if (colContainer) {
+    new Sortable(colContainer, {
+      group: 'cols',
+      animation: 150,
+      filter: '.ignore-elements',
+      // Element dragging ended
+      onEnd(/** Event*/ evt) {
+        const columns = [];
+        $(evt.to).each((_i, v) => {
```

Comment:
This seems like a case for `Object.values`, e.g. `for (const value of Object.values(e.to) {}`